### PR TITLE
[js/webgpu] Use DataType as uniform cpu type

### PR DESCRIPTION
**Pull request by axinging**
This saves turning data type to string by tensorDataTypeEnumToString.

### CODE REVIEW

**Comment by axinging** [comment]
@qjia7 @xhcao @hujiajie @gyagp PTAL

File: js/web/lib/wasm/jsep/backend-webgpu.ts [hunk 1]
```diff
@@ -458,11 +458,11 @@ export class WebGpuBackend {
...
+        if (v.type === DataType.int32) {
           new Int32Array(arrayBuffer, offset, data.length).set(data);
-        } else if (v.type === 'uint32') {
+        } else if (v.type === DataType.uint32) {
           new Uint32Array(arrayBuffer, offset, data.length).set(data);
-        } else if (v.type === 'float16') {
+        } else if (v.type === DataType.float16) {
           // TODO: use Float16Array.
           new Uint16Array(arrayBuffer, offset, data.length).set(data);
         } else {
```
[the remark this answers is not in the document]
Comment (axinging) [reply js/web/lib/wasm/jsep/backend-webgpu.ts]: done, PTAL

**Review by qjia7 — APPROVED** [review]
LGTM, thanks!

@fs-eire @guschmue @satyajandhyala Please take a look, thanks.

**Review by guschmue — APPROVED** [review]

**Comment by guschmue** [comment]
lint nagging, want you to run 'npm run format'

**Comment by axinging** [comment]
> lint nagging, want you to run 'npm run format'

Updated, PTAL

**Review by guschmue — DISMISSED** [review]

**Comment by guschmue** [comment]
Need to be merged with main

**Review by guschmue — APPROVED** [review]

**Pull request merged**